### PR TITLE
 feat(modal form submit): Change the way we submit the modal forms.

### PR DESCRIPTION
**Pull request by Camwyn**
Restructure the modal forms to one form.
Remove them from being nested inside the tickets block form.
Add a hidden field to contain all the data we were sending to the API - so the cart/checkout page can read it without having to change that code too much.

🎫 https://central.tri.be/issues/136609

Aside from one small style fix, this does not address any outstanding bugs.

requires/branched off of https://github.com/moderntribe/event-tickets/pull/1433

### CODE REVIEW

File: src/Tribe/Assets.php [hunk 1]
```diff
@@ -68,12 +68,11 @@ public function admin_enqueue_scripts() {
...
 
 		/**
-		 * Filter the decimal point character used in the price
-		 * @param string $decimal the decimal character to filter
+		 * Filter the decimal point character used in the price.
 		 *
 		 * @since 4.6
 		 *
-		 * @param string $decimal
+		 * @param string $decimal the decimal character to filter
```
Comment (sc0ttkclark) [inline src/Tribe/Assets.php]:
```suggestion
		 * @param string $decimal The decimal character to filter.
```

File: src/Tribe/Attendee_Registration/Modal.php [hunk 2]
```diff
@@ -14,7 +14,7 @@ class Tribe__Tickets__Attendee_Registration__Modal {
 	public function hook() {
 
 		add_filter( 'tribe_events_tickets_attendee_registration_modal_content', [ $this, 'modal_cart_template' ], 10, 2 );
-		add_filter( 'tribe_events_tickets_attendee_registration_modal_content', [ $this, 'append_modal_ar_template' ], 11, 2 );
+		// /add_filter( 'tribe_events_tickets_attendee_registration_modal_content', [ $this, 'append_modal_ar_template' ], 11, 2 );
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Modal.php]:
```suggestion
		// add_filter( 'tribe_events_tickets_attendee_registration_modal_content', [ $this, 'append_modal_ar_template' ], 11, 2 );
```

File: src/Tribe/Commerce/PayPal/Main.php [hunk 3]
```diff
@@ -2084,6 +2087,36 @@ public function rest_get_checkout_url( $checkout_url, $data, $post_id ) {
...
+	/**
+	 * Adds cart url to list used for localized variables.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $urls The original array.
+	 * @return array
+	 */
+	public function add_cart_url( $urls = [] ) {
+		$cart_url = '';
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/PayPal/Main.php]:
```suggestion
		$cart_url = site_url();
```

File: src/Tribe/Commerce/PayPal/Main.php [hunk 4]
```diff
@@ -2084,6 +2087,36 @@ public function rest_get_checkout_url( $checkout_url, $data, $post_id ) {
...
+	/**
+	 * Adds checkout url to list used for localized variables.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $urls The original array.
+	 * @return array
+	 */
+	public function add_checkout_url( $urls = [] ) {
+		$checkout_url = '';
```
Comment (sc0ttkclark) [inline src/Tribe/Commerce/PayPal/Main.php]:
```suggestion
		$checkout_url = site_url();
```

File: src/Tribe/Editor/Blocks/Tickets.php [hunk 5]
```diff
@@ -97,9 +96,29 @@ public function assets() {
 
 		wp_enqueue_script('wp-utils');
 
-		$ajax_preload               = apply_filters( 'tribe_tickets_preload_cart_in_ticket_form', false );
+		// $ticket_block_preload               = apply_filters( 'tribe_tickets_preload_cart_in_ticket_form', false );
 		$availability_check_interval = apply_filters( 'tribe_tickets_availability_check_interval', 0 );
 
+		$checkout_urls = [];
+		/**
```
Comment (sc0ttkclark) [inline src/Tribe/Editor/Blocks/Tickets.php]:
```suggestion

		/**
```

File: src/Tribe/Editor/Blocks/Tickets.php [hunk 6]
```diff
@@ -97,9 +96,29 @@ public function assets() {
...
+		 * Allow providers to add their own checkout URL to the localized list.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $checkout_urls An array to add urls to.
+		 */
+		$checkout_urls = apply_filters( 'tribe_tickets_checkout_urls', $checkout_urls );
+
+		$cart_urls = [];
+		/**
```
Comment (sc0ttkclark) [inline src/Tribe/Editor/Blocks/Tickets.php]:
```suggestion

		/**
```

File: src/Tribe/Editor/Blocks/Tickets.php [hunk 7]
```diff
@@ -132,7 +151,14 @@ public function assets() {
...
 						'data' => $this->set_messages(),
-					]
+					],
+					[
+						'name' => 'TribeTicketsURLs',
+						'data' => [
+							'cart'     => $cart_urls,
+							'checkout' => $checkout_urls,
+						]
+						],
```
Comment (sc0ttkclark) [inline src/Tribe/Editor/Blocks/Tickets.php]:
```suggestion
					],
```

File: src/Tribe/Editor/Blocks/Tickets.php [hunk 8]
```diff
@@ -132,7 +151,14 @@ public function assets() {
...
 						'name' => 'TribeMessages',
 						'data' => $this->set_messages(),
-					]
+					],
+					[
+						'name' => 'TribeTicketsURLs',
+						'data' => [
+							'cart'     => $cart_urls,
+							'checkout' => $checkout_urls,
+						]
```
Comment (sc0ttkclark) [inline src/Tribe/Editor/Blocks/Tickets.php]:
```suggestion
						],
```

File: src/resources/postcss/tickets.pcss [hunk 9]
```diff
@@ -507,6 +507,10 @@
 					background: var(--color-icon-error) ! important;
 				}
 			}
+
+			+ .tribe-ticket {
```
Comment (sc0ttkclark) [inline src/resources/postcss/tickets.pcss]: Is this intentional `+`?

File: src/views/blocks/tickets.php [hunk 10]
```diff
@@ -120,3 +120,6 @@
 		echo $html;
 	?>
 </form>
+<div  class="tribe-common">
```
Comment (sc0ttkclark) [inline src/views/blocks/tickets.php]:
```suggestion
<div class="tribe-common">
```

File: src/resources/postcss/tickets.pcss (hunk 9 above)
Comment (Camwyn) [reply src/resources/postcss/tickets.pcss]: yes it's `.tribe-ticket + .tribe-ticket` or "any `.tribe-ticket` following a `.tribe-ticket`"

**Pull request merged**